fuel-core-interfaces/src/txpool.rs

```diff
@@ -103,7 +104,9 @@ pub enum Error {
     #[error("Transaction is not inserted. ContractId is already taken {0:?}")]
     NotInsertedContractIdAlreadyTaken(ContractId),
     #[error("Transaction is not inserted. UTXO is not existing: {0:?}")]
-    NotInsertedInputTxNotExisting(TxId),
+    NotInsertedInputUtxoIdNotExisting(UtxoId),
+    #[error("Transaction is not inserted. UTXO is spent: {0:?}")]
```

Comment:
I saw we have tests for the other error messages, it could be helpful to add one for this one as well as the `fuel-core` progress to using utxo validation in more and more places, and it is used in a couple of places.

Reply:
Added test for new error, to check if we handle when coin is spent.